crates/craby_codegen/src/generators/rs_generator.rs

```diff
@@ -370,9 +487,56 @@ impl RsTemplate {
         let has_signals = ctx.schemas.iter().any(|schema| !schema.signals.is_empty());
         let rs_cxx_bridges = self.rs_cxx_bridges(&ctx.schemas)?;
         let cxx_impls = self.rs_cxx_impl(&rs_cxx_bridges);
-        let cxx_externs = self.rs_cxx_extern(&cxx_ns, &rs_cxx_bridges, has_signals);
+        let cxx_externs = self.rs_cxx_extern(&cxx_ns, &rs_cxx_bridges, has_signals, &ctx.schemas);
+        
+        // Generate signal payload extraction function implementation
+        let signal_payload_impls = if has_signals {
+            ctx.schemas.iter().flat_map(|schema| {
+                if schema.signals.is_empty() {
+                    return vec![];
+                }
+                
+                let signal_enum_name = format!("{}Signal", schema.module_name);
+                let mut impls: Vec<String> = schema.signals.iter().filter_map(|signal| {
+                    signal.payload_type.as_ref().map(|payload_type| {
+                        let payload_type_name = payload_type.as_rs_type()
+                            .map(|t| t.into_code())
+                            .unwrap_or_else(|_| "String".to_string());
+                        let function_name = format!("get_{}_payload", snake_case(&signal.name));
+                        let signal_variant = pascal_case(&signal.name);
+                        
+                        formatdoc! {
+                            r#"
+                            fn {function_name}(s: &{signal_enum_name}) -> {payload_type_name} {{
+                                match s {{
+                                    {signal_enum_name}::{signal_variant}(payload) => (*payload).clone(),
+                                    _ => panic!("Invalid signal type for {function_name}"),
+                                }}
+                            }}"#,
+                        }
+                    })
+                }).collect();
+                
+                // Add drop_signal implementation
+                impls.push(formatdoc! {
+                    r#"
+                    unsafe fn drop_signal(signal: *mut {signal_enum_name}) {{
+                        if !signal.is_null() {{
+                            let _ = Box::from_raw(signal);
+                        }}
```

Comment:
While it's true that the value will be implicitly dropped when it goes out of scope, what do you think about explicitly dropping it to make the intent clearer?

```suggestion
                        if !signal.is_null() {{
                            drop(Box::from_raw(signal));
                        }}
```